bravado/client.py

```diff
@@ -238,34 +232,74 @@ def __init__(self, name, operations):
         self._name = name
         self._operations = operations
 
+    # XXX 1.2
+    # @classmethod
+    # def from_api_doc(cls, api_doc, http_client, base_path, url_base=None):
+    #     """
+    #     :param api_doc: api doc which defines this resource
+    #     :type  api_doc: :class:`dict`
+    #     :param http_client: a :class:`bravado.http_client.HttpClient`
+    #     :param base_path: base url to perform api requests. Used to override
+    #             the path provided in the api spec
+    #     :param url_base: a url used as the base for resource definitions
+    #             that include a relative basePath
+    #     """
+    #     declaration = api_doc['api_declaration']
+    #     models = build_models(declaration.get('models', {}))
+    #
+    #     def build_operation(api_obj, operation):
+    #         log.debug(u"Building operation %s.%s" % (
+    #             api_obj.get('name'), operation['nickname']))
+    #
+    #         resource_base_path = declaration.get('basePath')
+    #         url = get_resource_url(base_path, url_base, resource_base_path)
+    #         url = url.rstrip('/') + api_obj['path']
+    #         return Operation(url, operation, http_client, models)
+    #
+    #     operations = dict(
+    #         (oper['nickname'], build_operation(api, oper))
+    #         for api in declaration['apis']
+    #         for oper in api['operations'])
+    #     return cls(api_doc['name'], operations)
```

Comment:
I'm planning to do the 'resource' stuff in a separate pull request. I named this branch incorrectly as 'port resource' instead of 'port model'  -- maybe the cause of confusion.